components/apimgt/org.wso2.carbon.apimgt.rest.api.endpoint.registry/src/main/java/org/wso2/carbon/apimgt/rest/api/endpoint/registry/impl/RegistriesApiServiceImpl.java

```diff
@@ -164,7 +164,23 @@ public Response createRegistryEntry(String registryId, RegistryEntryDTO registry
     @Override
     public Response updateRegistry(String registryId, RegistryDTO body, MessageContext messageContext) {
 
-        return Response.ok().entity(body).build();
+        String tenantDomain = RestApiUtil.getLoggedInUserTenantDomain();
+        String user = RestApiUtil.getLoggedInUsername();
+        EndpointRegistry registryProvider = new EndpointRegistryImpl();
+        EndpointRegistryInfo registryToUpdate = EndpointRegistryMappingUtils.fromDTOtoEndpointRegistry(body, user);
+        try {
+            EndpointRegistryInfo endpointRegistry =
+                    registryProvider.getEndpointRegistryByUUID(registryId, tenantDomain);
+            if (endpointRegistry == null) {
+                RestApiUtil.handleResourceNotFoundError("Endpoint registry with the id: " + registryId +
+                        " is not found", log);
+            }
+            registryProvider.updateEndpointRegistry(registryId, registryToUpdate);
+        } catch (APIManagementException e) {
+            RestApiUtil.handleInternalServerError("Error while updating the endpoint registry " +
+                    "with id: " + registryId, e, log);
+        }
+        return Response.ok().entity("Successfully updated the endpoint registry").build();
```

Comment:
Shall we return the updated Registry object in response.

Reply:
ack. missed it